### DISCUSSION TITLE
What is the correct way to obfuscate, bundle, and run a Fastify TypeScript project in a production environment?

### DISCUSSION THREAD

**Discussion by seepine** [fastify/fastify#5478] [Q&A]
This might not be a problem with Fastify, but I would like to ask how do it.

My current approach is
1. run `tsc -p tsconfig.json` to dist/app.js
2. run `javascript-obfuscator dist --output dist-obfuscated --source-map true`
3. run `cd dist-obfuscated` and `node app.js`
But cannot source map success when error happend, i got 
```shell
[2024-05-22 22:30:45.211 +0800] ERROR: [error-handler] TypeError: _0x364753[_0x25aee3(...)] is not a function
    at /root/workspace/fastify-template-ts/dist-obfuscated/src/sourceMap:58:21
    at Generator.next (<anonymous>)
    at /root/workspace/fastify-template-ts/dist-obfuscated/src/sourceMap:8:14
    at new Promise (<anonymous>)
```

I am trying to use [npm/source-map-support]( https://www.npmjs.com/package/source-map-support) and code `import 'source-map-support/register` at the top, but cannot work event.

**Comment by DataM0del** [comment]
Why would you?
